web/src/app/util/copyToClipboard.ts

```diff
@@ -13,25 +13,27 @@ export default function copyToClipboard(text) {
  * If other text was highlighted before this operation,
  * that state is saved upon completion of copying.
  */
-function fallback(str) {
-  const textArea = document.createElement('textArea')
+function fallback(str: string) {
+  const textArea = <HTMLTextAreaElement>document.createElement('textArea')
   textArea.value = str // Set its value to what you want copied
   textArea.readOnly = true // Deny tampering
   document.body.appendChild(textArea)
 
   // Check if there is any content selected previously
   const selected =
-    document.getSelection().rangeCount > 0
-      ? document.getSelection().getRangeAt(0) // Store selection if found
+    document.getSelection()!.rangeCount > 0
+      ? document.getSelection()!.getRangeAt(0) // Store selection if found
       : false
 
   // iOS requires some special finesse
   if (isOS()) {
     const range = document.createRange()
     range.selectNodeContents(textArea)
     const selection = window.getSelection()
-    selection.removeAllRanges()
-    selection.addRange(range)
+    if (selection) {
+      selection.removeAllRanges()
+      selection.addRange(range)
+    }
```

Comment:
If it fails to get the selection, does the copy still succeed? If not, let's return an error here if selection is falsey